File: backend/src/main/java/com/example/backend/dto/project/NewProjectDto.java
package com.example.backend.dto.project;

import com.example.backend.dto.customMilestone.CustomMilestonesDto;
import jakarta.annotation.Nullable;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

@AllArgsConstructor
@NoArgsConstructor
@Data
public class NewProjectDto {

    private String name;
    private Long ownerId;
    private List<Long> membersId;
    @Nullable
    private List<CustomMilestonesDto> customMilestones;
    private LocalDateTime deadline;
}

File: frontend/src/app/shared/dto/new-project.dto.ts
import { NewProjectMilestoneDto } from './new-project-milestone.dto';

export type NewProjectDto = {
  name: string;
  description: string;
  membersId: number[];
  customMilestones: NewProjectMilestoneDto[];
  deadline:string;
  ownerId:number;
};
